packages/modules/viewer-react/src/components/BaseViewer.tsx

```diff
@@ -55,24 +35,10 @@ export const BaseViewer: React.FC<ViewerProps> = ({
   const accessToken = useAccessToken();
   return (
     <ErrorBoundary>
-      {snapshotPath || accessToken ? (
+      {("filePath" in iModelLoaderProps && iModelLoaderProps.filePath) ||
```

Comment:
Do we need the first part of this check (`"filePath" in iModelLoaderProps`)? What if iModelLoaderProps is undefined for some reason - it would throw an error - although that would be unlikely it seems

Reply:
Yea this was really annoying, and probably the ugliest part of all my changes. I was hoping it could infer `filePath` existed from the union, but it couldn't auto infer it because of the union. Needed to use the `in` keyword to get the [discriminated union](https://www.typescriptlang.org/play?q=267#example/discriminate-types) to work. I need to see if I can clean this up with the `XOR`

Reply:
Looks like there is no other way to get around this than using the `"in" in loaderProps`  due to the union :/ . `loaderProps` should never be undefined due to the xor, it should always have the required props for the specific viewer (connected/blank/file)